api/app/src/command/webhook/webhook.ts

```diff
@@ -0,0 +1,249 @@
+  payloads: WebhookDataPayloadWithoutMessageId[]
+): Promise<boolean> => {
+  // TODO #43 Check if there are failed chunks from previous attempts ans start with those
+  for (const payload of payloads) {
+    const success = await processChunk(cxId, payload, settings);
+    // give it some time to prevent flooding the customer
+    if (success) await Util.sleep(Math.random() * 200);
+  }
+  return true;
+};
+
+const processChunk = async (
+  cxId: string,
+  payload: WebhookDataPayloadWithoutMessageId,
+  settings: Settings
+): Promise<boolean> => {
+  // create a representation of this request and store on the DB
+  const webhookRequest = await createWebhookRequest({ cxId, payload });
+
+  const { webhookUrl, webhookKey, webhookEnabled } = settings;
+  if (!webhookUrl || !webhookKey) {
+    console.log(
+      `Missing webhook config, skipping sending it ` +
+        `(url: ${webhookUrl}, key: ${webhookKey ? "<defined>" : null}`
+    );
+    return false;
+  }
+  // TODO Separate preparing the payloads with the actual sending of that data over the wire
+  // It will simplify managing the webhook status (enabled?) and only retrying sending once
+  // every X minutes
+  try {
+    await sendPayload(
+      {
+        meta: {
+          messageId: webhookRequest.id,
+          when: dayjs(webhookRequest.createdAt).toISOString(),
+        },
+        ...payload,
+      },
+      webhookUrl,
+      webhookKey
+    );
+    // mark this request as successful on the DB
+    await updateWebhookRequestStatus({
+      id: webhookRequest.id,
+      status: "success",
+    });
+
+    // if the webhook was not working before, update the status to successful since we were able to send the payload
+    if (!webhookEnabled) {
+      await updateWebhookStatus({
+        id: settings.id,
+        webhookEnabled: true,
+        webhookStatusDetail: WEBHOOK_STATUS_OK,
+      });
+    }
+    return true;
+  } catch (err: any) {
+    try {
+      // mark this request as failed on the DB
+      await updateWebhookRequestStatus({
+        id: webhookRequest.id,
+        status: "failure",
+      });
+    } catch (err2) {
+      console.log(`Failed to store failure state on WH log`, err);
+    }
+    let webhookStatusDetail;
+    if (err instanceof WebhookError) {
+      webhookStatusDetail = err.underlyingError.message;
+    } else {
+      log(`Unexpected error testing webhook`, err);
+      webhookStatusDetail = `Internal error: ${err?.message}`;
+    }
+    try {
+      // update the status of this webhook on the DB
+      await updateWebhookStatus({
+        id: settings.id,
+        webhookEnabled: false,
+        webhookStatusDetail,
+      });
+    } catch (err2) {
+      console.log(`Failed to store failure state on WH settings`, err);
+    }
+  }
+  return false;
+};
+
+const sendPayload = async (
+  payload: WebhookPayload,
+  url: string,
+  apiKey: string,
+  timeout = 5_000
+): Promise<any> => {
+  log(`Sending payload:\n${JSON.stringify(payload)}`);
```

Comment:
Is this for debugging purposes? Can blow up our logs fairly quickly

Reply:
oh yeah, I'll remove it, good catch!